src/monexamine.cpp

```diff
@@ -313,6 +321,60 @@ bool monexamine::mech_hack( monster &z )
     return false;
 }
 
+static int prompt_for_amount( const char *const msg, const int max )
+{
+    const std::string formatted = string_format( msg, max );
+    const int amount = string_input_popup()
+                       .title( formatted )
+                       .width( 20 )
+                       .text( to_string( max ) )
+                       .only_digits( true )
+                       .query_int();
+
+    return ( amount > max ) ? max : ( amount <= 0 ) ? 0 : amount;
+}
+
+bool monexamine::pay_bot( monster &z )
+{
+    time_duration friend_time = z.get_effect_dur( effect_pet );
+    const int card_count = g->u.amount_of( "cash_card" );
+    const int charge_count = card_count ? g->u.charges_of( "cash_card" ) : 0;
+
+    int amount = 0;
+    uilist bot_menu;
+    bot_menu.text = string_format(
+                        _( "Welcome to the %s Friendship Interface. What would you like to do?\n"
+                           "Your current friendship will last: %s" ), z.get_name(), to_string( friend_time ) );
+    if( charge_count > 0 ) {
+        bot_menu.addentry( 1, true, 'b', _( "Get more friendship. 10 cents/min" ) );
+    } else {
+        bot_menu.addentry( 2, true, 'q',
+                           _( "Sadly you're not currently able to extend your friendship. - Quit menu" ) );
+    }
+    bot_menu.query();
+    switch( bot_menu.ret ) {
+        case 1:
+            amount = prompt_for_amount( ngettext(
+                                            "How much friendship do you get? Max: %d minute. (0 to cancel) ",
+                                            "How much friendship do you get? Max: %d minutes. ", charge_count / 10 ), charge_count / 10 );
+            if( amount > 0 ) {
+                time_duration time_bought = time_duration::from_minutes( amount );
```

Comment:
Won't this be a cent/minute?
```suggestion
                time_duration time_bought = time_duration::from_minutes( amount / 6 );
```

Reply:
You actually pay at the line below ` g->u.use_charges( "cash_card", amount * 10 );`

Reply:
Oh, my bad!